### ISSUE TITLE
Kernel mismatch between docker 4.4.2 and docker/for-desktop-kernel image

### ISSUE THREAD

**Issue by Molter73** [docker/for-mac#6153]
- [x] I have tried with the latest version of Docker Desktop
  - [ ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID:

### Expected behavior
Building a kernel modules using the kernel headers provided in the [5.10.76 image](https://hub.docker.com/layers/docker/for-desktop-kernel/5.10.76-505289bcc85427a04d8d797e06cbca92eee291f4-amd64/images/sha256-3de6fb91d266901a96e72c01d42db9f8aef7fd81cc6193abef07a1c58f8ce699?context=explore) creates a module that properly works with docker desktop 4.4.2
### Actual behavior
Inserting the compiled module fails with `Exec format error`
### Information

  - macOS Version: 11.6.1
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.4.2

The problem seems to be a mismatch between the supplied kernel headers in the image and the actual kernel running in docker for desktop. Extracting the kernel banner from the binary in the image gives:

```
[root@cf34eed57fcb out]# skip="$(LC_ALL=C grep -a -b -o $'\x1f\x8b\x08\x00\x00\x00\x00\x00' kernel | head -n 1 | cut -d ':' -f 1)" 
[root@cf34eed57fcb out]# dd if=kernel bs=1 skip="$skip" | zcat -q | tr -d '\0' | grep -a 'Linux version [0-9]' || true
Linux version 5.10.76-linuxkit (root@buildkitsandbox) (gcc (Alpine 10.2.1_pre1) 10.2.1 20201203, GNU ld (GNU Binutils) 2.35.2) #1 SMP Tue Nov 9 08:49:36 UTC 2021
```

But running `uname -v` from a container running in docker 4.4.2 outputs:
```
#1 SMP Mon Nov 8 10:21:19 UTC 2021
```

So the kernel provided with the docker headers and the one in the image have been compiled at different times.

### Steps to reproduce the behavior

  1. Build a kernel module using the headers in the [5.10.76 image](https://hub.docker.com/layers/docker/for-desktop-kernel/5.10.76-505289bcc85427a04d8d797e06cbca92eee291f4-amd64/images/sha256-3de6fb91d266901a96e72c01d42db9f8aef7fd81cc6193abef07a1c58f8ce699?context=explore).
  2. Launch a privileged container and attempt to load the module.

**Comment by Molter73** [comment]
This issue persists in version 4.5.0.

Based on this repo: https://github.com/djs55/docker-desktop-example-kernel-module. I made some changes to the dockerfile, compiled the example module and attempted to mount it.

```
mmoltras in ~/go/src/github.com/djs55/docker-desktop-example-kernel-module/kernel on main ● λ docker build -t kernel-module .
mmoltras in ~/go/src/github.com/djs55/docker-desktop-example-kernel-module/kernel on main ● λ docker run -it --rm --privileged kernel-module:latest
/kmod # insmod exmple.ko
insmod: can't insert 'exmple.ko': No such file or directory
/kmod # dmesg
[...]
[  180.616001] example: version magic '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_86e8285ec1eb0c8509f422448207017c33f8fe32df12be4e401a2894aa6daf47' should be '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_974f2e0f9b2cfe01cccdb3a1126d8a32b9f02ccaaa27f2cbc1d452a055348503'
[  180.630745] example: version magic '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_86e8285ec1eb0c8509f422448207017c33f8fe32df12be4e401a2894aa6daf47' should be '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_974f2e0f9b2cfe01cccdb3a1126d8a32b9f02ccaaa27f2cbc1d452a055348503'
[  320.164773] example: version magic '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_86e8285ec1eb0c8509f422448207017c33f8fe32df12be4e401a2894aa6daf47' should be '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_974f2e0f9b2cfe01cccdb3a1126d8a32b9f02ccaaa27f2cbc1d452a055348503'
[  320.179521] example: version magic '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_86e8285ec1eb0c8509f422448207017c33f8fe32df12be4e401a2894aa6daf47' should be '5.10.76-linuxkit SMP mod_unload modversions RANDSTRUCT_PLUGIN_974f2e0f9b2cfe01cccdb3a1126d8a32b9f02ccaaa27f2cbc1d452a055348503'
```

Here is the modified dockerfile I used:
```Dockerfile
# This Dockerfile extracts the kernel headers from the kernel image
# and then compiles a simple hello world kernel module against them.
# In the last stage, it creates a package, which can be used for
# testing.

FROM docker/for-desktop-kernel:5.10.76-505289bcc85427a04d8d797e06cbca92eee291f4-amd64 AS ksrc

# Extract headers and compile module
FROM linuxkit/alpine:2be490394653b7967c250e86fd42cef88de428ba AS build
RUN apk add build-base elfutils-dev

COPY --from=ksrc /kernel-dev.tar /
RUN tar xf kernel-dev.tar

WORKDIR /kmod
COPY ./* ./
RUN make all
```

No other changes where made to the repository.
Running the same commands on v4.2.0 (and using the appropriate `docker/for-desktop-kernel` image) works correctly.

**Comment by Molter73** [comment]
This issue seems to have been fixed around v4.6.0 and is not present in 4.7 or 4.8, so I'm closing this issue.
